### PR TITLE
updated torch implementation of z2color copy with momentum, lr decay,…

### PR DESCRIPTION
**Pull request by BaladityaY**
… weight decay, etc.

### CODE REVIEW

File: Train.py [hunk 1]
```diff
@@ -31,9 +31,23 @@ def main():
     torch.cuda.set_device(ARGS.gpu)
     torch.cuda.device(ARGS.gpu)
 
-    net = SqueezeNetShallow().cuda()
+    net = Z2Bala().cuda()
     criterion = torch.nn.MSELoss().cuda()
-    optimizer = torch.optim.Adagrad(net.parameters(), lr=.1)
+    
+    base_lr = .005 # z2caffe = .005
```
Comment (raunakdoesdev) [inline Train.py:37]: Please read the PyTorch docs on optimizers, they already do all of what was implemented here in a much more efficient manner (in C code rather than Python). Find these docs here:

http://pytorch.org/docs/master/optim.html

File: nets/Z2Bala.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.nn.init as init
+from torch.autograd import Variable
+import logging
+logging.basicConfig(filename='training.log', level=logging.DEBUG)
+
+from Parameters import ARGS
+
+class Fire(nn.Module):
```
Comment (raunakdoesdev) [inline nets/Z2Bala.py:12]: Fire module insertion here is not necessary.

File: nets/Z2Bala.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+"""SqueezeNet 1.1 modified for regression."""
```
Comment (raunakdoesdev) [inline nets/Z2Bala.py:1]: Let's try to keep the doc strings accurate.

File: nets/Z2Bala.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import torch
```
Comment (raunakdoesdev) [inline nets/Z2Bala.py:2]: This entire file is a duplicate of:
https://github.com/bddmodelcar/training/blob/master/nets/Z2ColorBatchNorm.py
except w/o the BatchNorm is it not? unless there is some important reason for removing this BatchNorm why did you create this file?

**Review by raunakdoesdev — COMMENTED** [review]
Made some interesting changes, but it definitely needs some more review before it is adopted into master. Most of the changes here are already implemented in the SGD optimizer, so I would suggest looking into that as it would clean up this code quite a bit. Thanks for contributing to the repo!
